app/validators/candidate_interface/continuous_applications/course_selection_validator.rb

```diff
@@ -3,20 +3,39 @@ module ContinuousApplications
     # Validates that the course choice is not duplicated
     # Conditional exists to remove the course that is being edited from the checks for duplication
     class CourseSelectionValidator < ActiveModel::Validator
+      ALLOWED_REAPPLICATION_LIMIT = 2
+
       def validate(record)
         return unless record.wizard.current_application
 
-        scope = record.wizard.current_application.application_choices.joins(:course_option)
-        scope = omit_current_application_choice(scope, record) if editing?(record)
-        exists = scope
-          .where({ status: ApplicationStateChange.non_reapply_states })
-                  .exists?(course_option: { course_id: record.course.id })
+        scope = scope_for_current_application(record)
+
+        if reached_reapplication_limit?(scope, record)
+          record.errors.add :base, :reached_reapplication_limit, message: 'You cannot apply to this training provider and course again'
```

Comment:
I thought about that, but I'm not sure it serves a purpose, apart from not making an extra query.

Reply:
Maybe use `elsif`

Reply:
Added a `return` in https://github.com/DFE-Digital/apply-for-teacher-training/pull/8747/commits/54a3c292bba492d0ff6a538fe73afd4320c51428

I wasn't sure if I wanted to do this because I feel the validator shouldn't be aware of the logic we require outside of it. But instead to return all the errors. But I guess we will never need to check for duplicated courses in this case.